astropy/config/tests/test_configs.py

```diff
@@ -82,20 +82,17 @@ def test_func():
 
     # Test context manager mode (with cleanup)
     with paths.set_temp_cache(temp_cache_dir, delete=True):
-        assert paths.get_cache_dir(rootname='astropy') == temp_astropy_cache
+        assert paths.get_cache_dir(rootname="astropy") == str(temp_astropy_cache)
 
-    assert not os.path.exists(temp_cache_dir)
+    assert not temp_cache_dir.exists()
 
 
-def test_set_temp_cache_resets_on_exception(tmpdir):
+def test_set_temp_cache_resets_on_exception(tmp_path):
     """Test for regression of  bug #9704"""
     t = paths.get_cache_dir()
-    a = tmpdir / 'a'
-    with open(a, 'wt') as f:
-        f.write("not a good cache\n")
-    with pytest.raises(OSError):
-        with paths.set_temp_cache(a):
-            pass
+    (a := tmp_path / "a").write_text("not a good cache\n")
+    with pytest.raises(OSError), paths.set_temp_cache(a):
```

Comment:
This is slightly more confusing because it is less obvious that the first context manager is needed for the second one but I guess functionally, it doesn't matter.